Kernel/Output/HTML/Preferences/ColumnFilters.pm

```diff
@@ -57,14 +59,28 @@ sub Run {
 
     return 1 if !defined $FilterAction;
 
-    for my $Key ( sort keys %{ $Param{GetParam} } ) {
+    if ( !$Kernel::OM->Get('Kernel::Config')->Get('DemoSystem') ) {
+        KEY:
+        for my $Key ( sort keys %{ $Param{GetParam} } ) {
 
-        # pref update db
-        if ( !$Kernel::OM->Get('Kernel::Config')->Get('DemoSystem') ) {
+            # We require non-empty list of column names - ignore antyhing else.
+            next KEY if !( $Param{GetParam}->{$Key} && IsArrayRefWithData( $Param{GetParam}->{$Key} ) );
```

Comment:
```suggestion
            next KEY if !$Param{GetParam}->{$Key};
            next KEY if !IsArrayRefWithData($Param{GetParam}->{$Key});
```

Reply:
Done in 4252343ca6f6cdf84d540a9c08635c414511d75b.